Brokerages/SymbolPropertiesDatabaseSymbolMapper.cs

```diff
@@ -0,0 +1,182 @@
+                SymbolPropertiesDatabase
+                    .FromDataFolder()
+                    .GetSymbolPropertiesList(_market)
+                    .Where(x => !string.IsNullOrWhiteSpace(x.Value.BrokerageSymbol))
+                    .ToList();
+
+            _symbolPropertiesMap = 
+                symbolPropertiesList
+                    .ToDictionary(x => x.Key, x => x.Value);
+
+            _symbolMap =
+                symbolPropertiesList
+                    .ToDictionary(x => x.Value.BrokerageSymbol, x => x.Key);
+        }
+
+        /// <summary>
+        /// Converts a Lean symbol instance to a brokerage symbol
+        /// </summary>
+        /// <param name="symbol">A Lean symbol instance</param>
+        /// <returns>The brokerage symbol</returns>
+        public string GetBrokerageSymbol(Symbol symbol)
+        {
+            if (symbol == null || string.IsNullOrWhiteSpace(symbol.Value))
+            {
+                throw new ArgumentException($"Invalid symbol: {(symbol == null ? "null" : symbol.Value)}");
+            }
+
+            if (symbol.ID.Market != _market)
+            {
+                throw new ArgumentException($"Invalid market: {symbol.ID.Market}");
+            }
+
+            var key = new SecurityDatabaseKey(symbol.ID.Market, symbol.Value, symbol.SecurityType);
+
+            SymbolProperties symbolProperties;
+            if (!_symbolPropertiesMap.TryGetValue(key, out symbolProperties) )
+            {
+                throw new ArgumentException($"Unknown symbol: {symbol.Value}/{symbol.SecurityType}/{symbol.ID.Market}");
+            }
+
+            if (string.IsNullOrWhiteSpace(symbolProperties.BrokerageSymbol))
+            {
+                throw new ArgumentException($"Brokerage symbol not found in database for symbol: {symbol.Value}");
+            }
+
+            return symbolProperties.BrokerageSymbol;
+        }
+
+        /// <summary>
+        /// Converts a brokerage symbol to a Lean symbol instance
+        /// </summary>
+        /// <param name="brokerageSymbol">The brokerage symbol</param>
+        /// <param name="securityType">The security type</param>
+        /// <param name="market">The market</param>
+        /// <param name="expirationDate">Expiration date of the security(if applicable)</param>
+        /// <param name="strike">The strike of the security (if applicable)</param>
+        /// <param name="optionRight">The option right of the security (if applicable)</param>
+        /// <returns>A new Lean Symbol instance</returns>
+        public Symbol GetLeanSymbol(string brokerageSymbol, SecurityType securityType, string market, DateTime expirationDate = default(DateTime), decimal strike = 0, OptionRight optionRight = OptionRight.Call)
+        {
+            if (string.IsNullOrWhiteSpace(brokerageSymbol))
+            {
+                throw new ArgumentException($"Invalid brokerage symbol: {brokerageSymbol}");
+            }
+
+            if (market != _market)
```

Comment:
Don't think we should send in the market here since it's already sent in the constructor, it doesn't seem like a real argument.
Edit: ah I see now it's in the interface, maybe once we migrate all mappers to use this new system we could remove it as an argument